ci/Jenkinsfile.android

```diff
@@ -49,6 +49,7 @@ pipeline {
         script {
           utils.symlinkEnv()
           println("Build Number: ${utils.genBuildNumber()}")
+          sh 'make generate-autolink-android'
```

Comment:
Why are we doing this here? It seems kinda awkward do do this as part of `Makefile` if it's always necessary for Android.

Why can't this be part of the Gradle build process for example? Is there ever a case where it don't need to run?

Reply:
We do this in 2 places, here for PR builds and inside `scripts/build-android.sh` for Debug builds which are run when we do `make run-android`.

The autolinking json file was introduced originally as a gradle build step by react-native team.
However the issue with that is this autolinking stuff runs `npx @react-native-community/cli config` at runtime and we can't run this inside our nix derivation since that runs in a sandbox environment. 

Hence I have made these changes.
I have also patched out this behaviour from react-native to make sure it works well with our build process.

The second question about is there ever a case where it doesn't need to be run, well we only need the autolinking json file right before we have to do an android build. Otherwise we don't need it.

Reply:
As discussed, i think we can probably generate that JSON file from `node_modules` in nix.
We should also ask RN devs what's the reasoning behind this step and why it needs the network. Examining the codebase might also shed some light on what they are trying to do.